src/main/scala/com/ebiznext/comet/utils/Job.scala

```diff
@@ -64,10 +65,32 @@ trait JobBase extends StrictLogging with DatasetLogging {
 
 trait SparkJob extends JobBase {
 
-  def withExtraSparkConf(): Map[String, String] = Map.empty
+  def withExtraSparkConf(sourceConfig: SparkConf): SparkConf = {
+    // During Job execution, schema update are done on the table before data is written
+    // These two options below are thus disabled.
+    // We disable them because even though the suer asked for WRITE_APPEND
+    // On merge, we write in WRITE_TRUNCATE mode.
+    sourceConfig.remove("spark.datasource.bigquery.allowFieldAddition")
+    sourceConfig.remove("spark.datasource.bigquery.allowFieldRelaxation")
+    sourceConfig
+  }
+
+  // Specific to BigQuery
+  val allowFieldAddition: Boolean =
+    if (settings.sparkConfig.hasPath("datasource.bigquery.allowFieldAddition"))
+      settings.sparkConfig.getString("datasource.bigquery.allowFieldAddition").toBoolean
+    else
+      false
```

Comment:
IMO, this should be in `BigQueryJobBase` instead (with an extends on `JobBase`).
And this way you can also check the BQ sink option:
```suggestion
      cliConfig.options.get("allowFieldAddition").exists(_.toBoolean)
```
Same for allowFieldRelaxation

Reply:
The Spark session is created in IngestionJob which is a direct subclass of SparkJob. That's why we set it here.
I am also going to handle field addition / relaxation without the user setting any option wince we are governed by the YAML file.

Reply:
But there are some issues with that choice.
- That you have to comment with `Specific to BigQuery` proves that should not be in the common interface
- You are defining a new default behaviour that will be inherited by all kind of spark jobs, but is highly specific to bigquery.  
- You don't have access to the `cliOptions#options` map, which can also be used by the end-user to configure column update.

Reply:
Actually, it's not a new behaviour.
It's simply that the YAML file is the reference for us. So if the user defined an updated YAML, this means that he wants the field addition / relaxation we are doing.